miio/yeelight_dual_switch.py

```diff
@@ -0,0 +1,233 @@
+
+    @property
+    def flex_mode(self) -> int:
+        return self.data["flex_mode"]
+
+    @property
+    def rc_list(self) -> str:
+        return self.data["rc_list"]
+
+    def __repr__(self) -> str:
+        s = (
+            "<YeelightDualControlModuleStatus "
+            "switch_1_status=%s, "
+            "switch_1_default_state=%s, "
+            "switch_1_delay=%s, "
+            "switch_2_status=%s, "
+            "switch_2_default_state=%s, "
+            "switch_2_delay=%s, "
+            "interlock=%s, "
+            "flex_mode=%s, "
+            "rc_list=%s>"
+            % (
+                self.switch_1_status,
+                self.switch_1_default_state,
+                self.switch_1_delay,
+                self.switch_2_status,
+                self.switch_2_default_state,
+                self.switch_2_delay,
+                self.interlock,
+                self.flex_mode,
+                self.rc_list,
+            )
+        )
+        return s
+
+
+class YeelightDualControlModule(MiotDevice):
+    """Main class representing the air humidifier which uses MIoT protocol."""
+
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+    ) -> None:
+        super().__init__(_MAPPING, ip, token, start_id, debug, lazy_discover)
+
+    @command(
+        default_output=format_output(
+            "",
+            "First Switch Status: {result.switch_1_status}\n"
+            "First Switch Default State: {result.switch_1_default_state}\n"
+            "First Switch Delay: {result.switch_1_delay}\n"
+            "Second Switch Status: {result.switch_2_status}\n"
+            "Second Switch Default State: {result.switch_2_default_state}\n"
+            "Second Switch Delay: {result.switch_2_delay}\n"
+            "Interlock: {result.interlock}\n"
+            "Flex Mode: {result.flex_mode}\n"
+            "RC list: {result.rc_list}\n",
+        )
+    )
+    def status(self) -> DualControlModuleStatus:
+        """Retrieve properties"""
+        p = [
+            "switch_1_status",
+            "switch_1_default_state",
+            "switch_1_delay",
+            "switch_2_status",
+            "switch_2_default_state",
+            "switch_2_delay",
+            "interlock",
+            "flex_mode",
+            "rc_list",
+        ]
+        properties = [
+            {"did": k, **v}
+            for k, v in filter(lambda item: item[0] in p, self.mapping.items())
+        ]
+        values = self.get_properties(properties, max_properties=1)
+        return DualControlModuleStatus(
+            dict(map(lambda v: (v["did"], v["value"]), values))
+        )
+
+    @command(
+        click.argument("switch_1_status", type=bool),
+        default_output=format_output("Set First Switch Status to: {switch_1_status}"),
+    )
+    def set_switch_1_status(self, switch_1_status: bool):
```

Comment:
I'm wondering if it'd be better to follow the naming scheme of other switch devices (i.e., use `on()` and `off()` for state changes), and add a parameter describing on which switch to actuate?

Reply:
Same applies to other methods too, not hardcoding these would allow extension in case a new switch comes up with three switches.

Reply:
good idea

Reply:
What about status? Leave it as it is?

Reply:
Just an idea :-) If done so, the class should expose the number of switches as a property to allow downstream users to figure out how many switches there exist.

Reply:
What do you suggest to use as parameter for choosing switch, enum or simply int
```python
class Switch(enum.Enum):
    First = 0
    Second = 1

def on(self, switch: Switch):

def on(self, switch: int):
    
```

Reply:
Both are fine by me, however you prefer. Maybe enum is nicer, just in case some future devices will have other types of switches?